### PR TITLE
Remove IServer and IBlockchain references in order to prevent Circular references

### PR DESCRIPTION
**Pull request by aboimpinto**
In order to sign the assets it's necessary to create the OperationManagers instead of been injected.

Classes like TransactionOperationManager will need to have more dependencies to perform the verification of the transaction and instead of create more singletons, we should just inject through the constructor.

The tests for the Genesis block creation need to be refactor.

### CODE REVIEW

**Comment by aboimpinto** [comment]
I had to change the way we create the genesis block in order to inject the TransactionOperationManager, BinarSerializer, Crypto instead of using the Singleton versions

**Review by osmirnov — COMMENTED** [review]
I disagree with IModel based classes and IGenesis like classes. I think IBlockModel is actually IBlockRepository because it just queries the persistence. IGenesis has to be IGenesisBuilder because what is doing is building GenesisBlock.

**Comment by aboimpinto** [comment]
I will rename this classes...

File: src/NeoSharp.Application/Controllers/PromptBlockchainController.cs [hunk 1]
```diff
@@ -17,6 +18,9 @@ public class PromptBlockchainController : IPromptController
         private readonly IBlockPool _blockPool;
         private readonly ITransactionPool _transactionPool;
         private readonly IBlockchain _blockchain;
+        private readonly IBlockRepository _blockModel;
```
Comment (osmirnov) [inline src/NeoSharp.Application/Controllers/PromptBlockchainController.cs]: all vars are still contain a model in their name

File: src/NeoSharp.Core/Blockchain/Genesis/GenesisAssets.cs [hunk 2]
```diff
@@ -10,24 +10,22 @@
 using NeoSharp.Core.Types;
 using NeoSharp.VM;
 
-namespace NeoSharp.Core.Blockchain
+namespace NeoSharp.Core.Blockchain.Genesis
 {
 #pragma warning disable CS0612 // Type or member is obsolete
-    public class GenesisAssets
+    public class GenesisAssets : IGenesisAssets
```
Comment (osmirnov) [inline src/NeoSharp.Core/Blockchain/Genesis/GenesisAssets.cs]: Do we consider this as a builder as well?

File: src/NeoSharp.Core/Blockchain/Genesis/IGenesisBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using NeoSharp.Core.Models;
+
+namespace NeoSharp.Core.Blockchain.Genesis
+{
+    public interface IGenesisBuilder
+    {
+        Block GenesisBlock { get; }
```
Comment (osmirnov) [inline src/NeoSharp.Core/Blockchain/Genesis/IGenesisBuilder.cs]: The Builder pattern assumes some method to output of the result: GenesisBlock Build(). Does it make sense?

**Pull request merged**